lighthouse-core/audits/lazy-third-party.js

```diff
@@ -0,0 +1,189 @@
+  failureTitle: 'Some third-party resources can be lazy loaded',
+  /** Description of a Lighthouse audit that identifies the third party code on the page that can be lazy loaded. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. Lazy loading means loading resources is deferred until they are needed. */
+  description: 'Some third-party resources can be fetched after the page loads. ' +
+    'These third-party resources are used by embedded elements which can be replaced by a facade ' +
+    'until the user needs to use them. [Learn more](https://web.dev/efficiently-load-third-party-javascript/).',
+  /** Summary text for the result of a Lighthouse audit that identifies the third party code on a web page that can be lazy loaded. This text summarizes the number of lazy loading facades that can be used on the page. Lazy loading means loading resources is deferred until they are needed. */
+  displayValue: `{itemCount, plural,
+  =1 {# facade alternative available}
+  other {# facade alternatives available}
+  }`,
+  /** Label for a table column that displays the name of a lazy loading facade alternative for a third party resource. Lazy loading means loading resources is deferred until they are needed. */
+  columnFacade: 'Facade Alternative',
+  /** Label for a table column that displays the name of the third party product that a URL is used for. Lazy loading means loading resources is deferred until they are needed. */
+  columnProduct: 'Product',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/** @typedef {import("third-party-web").IEntity} ThirdPartyEntity */
+/** @typedef {import("third-party-web").IProduct} ThirdPartyProduct*/
+
+/** @typedef {{product: ThirdPartyProduct, cutoffTime: number, urlSummaries: Map<string, ThirdPartySummary.Summary>}} ProductSummary */
+
+class LazyThirdParty extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'lazy-third-party',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs', 'URL'],
+    };
+  }
+
+  /**
+   * @param {Map<string, ThirdPartySummary.Summary>} byURL
+   * @param {ThirdPartyEntity | undefined} mainEntity
+   * @return {ProductSummary[]}
+   */
+  static getProductSummaries(byURL, mainEntity) {
+    /** @type {Map<string, Map<string, ProductSummary>>} */
+    const entitySummaries = new Map();
+
+    for (const [url, urlSummary] of byURL) {
+      const entity = thirdPartyWeb.getEntity(url);
+      const product = thirdPartyWeb.getProduct(url);
+      if (!entity || mainEntity && entity.name === mainEntity.name) continue;
```

Comment:
I think there's a helper function for this already, `isThirdParty`, which me and @patrickhulce planned on putting in a common place I think